lib/spack/spack/build_systems/generic.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from typing import Tuple
+
+import spack.builder
+import spack.directives
+import spack.package_base
+
+from ._checks import BaseBuilder, apply_macos_rpath_fixups
+
+
+class Package(spack.package_base.PackageBase):
+    """General purpose class with a single ``install`` phase that needs to be
+    coded by packagers.
+    """
+
+    #: This attribute is used in UI queries that require to know which
+    #: build-system class we are using
+    build_system_class = "Package"
+    #: Legacy buildsystem attribute used to deserialize and install old specs
+    legacy_buildsystem = "generic"
+
+    spack.directives.build_system("generic")
+
+
+@spack.builder.builder("generic")
+class GenericBuilder(BaseBuilder):
```

Comment:
Seeing some inconsistencies here:

* PackageBase vs. BaseBuilder
* Package vs. GenericBuilder

Would be nice if these were more consistent.

Reply:
`Package` is mandated by backward compatibility